src/main/java/com/google/devtools/build/lib/remote/AbstractRemoteActionCache.java

```diff
@@ -265,7 +265,7 @@ public void onFailure(Throwable t) {
           execRoot.getRelative(file.getPath()).delete();
         }
         for (OutputDirectory directory : result.getOutputDirectoriesList()) {
-          FileSystemUtils.deleteTree(execRoot.getRelative(directory.getPath()));
+          FileSystemUtils.deleteTreesBelow(execRoot.getRelative(directory.getPath()));
```

Comment:
could you add a comment above this line explaining why we're deleting the contents, and not the actual output directory? so that future devs don't change this back to `deleteTree`

Reply:
Done